### ISSUE TITLE
aksv2 - updating NP's multiple specs at once, results in Failed Provisioning state of NP

### ISSUE THREAD

**Issue by prabalsharma** [rancher/rancher#33049]
**What kind of request is this (question/bug/enhancement/feature request):** Bug

**Steps to reproduce (least amount of steps as possible):**
1. Create an AKS cluster with 2 NPs (1 system and 1 user). Autoscaler disabled, k8s version 1.19
2. Wait for it to become Active
3. **Edit** --> upgrade CP's  k8s version to 1.20 --> Save
4. Wait for the update to finish
5. **Edit**
6. Upgrade NP's k8s version to 1.20 (both System and User)
7. Enable Autoscaler for both NPs
8. Increment node count of both NPs
9. Save
10. Go to Azure and check NPs Provisioning State

**Result:**
Provisioning State my stay in Failed or Updating state. Even if update succeeds, it will take a very long time.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA):

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
(paste the output here)
```

- Docker version (use `docker version`):

```
(paste the output here)
```

**Comment by mjura** [comment]
I have tried to reproduce this issue and everything went fine.
1. I used latest Rancher and aks-operator master build
2. I used Rancher UI
3. In procedure I have upgraded Kubernete cluster version first then I went with NP upgrading. 
4. Changing or uprading two Node Pools at once should not be issue.

**Comment by prabalsharma** [comment]
I was able to repro on RC7 as well,

1. Rancher UI → Add cluster
2. hosted Kubernetes provider → Azure AKS 
3. Initial node count 1
4. Add NP → user Mode NP : node count 4
5. Create
6. Active → Edit
7. select System Mode  NP: enable autoscaling, node count to 4
8. select User Mode NP: enable autoscaling, node count to 2
9. Save

**Comment by mjura** [comment]
> I was able to repro on RC7 as well,
> 
> 1. Rancher UI → Add cluster
> 2. hosted Kubernetes provider → Azure AKS
> 3. Initial node count 1
> 4. Add NP → user Mode NP : node count 4
> 5. Create
> 6. Active → Edit
> 7. select System Mode  NP: enable autoscaling, node count to 4
> 8. select User Mode NP: enable autoscaling, node count to 2
> 9. Save

Can you please provide me info on what minCount and maxCount have you provided for System and User NP ?

**Comment by mjura** [comment]
Ok, I know what is wrong, Azure API can't handle enabling autoscaling and changing node count at this same time. This should be validated.

**Comment by mjura** [comment]
I have found also another issue, when disabling autoscaling for node pool, following error occurs:

```This cluster is currently Updating.

Failed to update cluster: containerservice.AgentPoolsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter" Message="AgentPool 'master' has 'agentPoolProfile.maxCount' set which requires that 'agentPoolProfile.enableAutoscaling' is true and it is false"
```

**Comment by mjura** [comment]
Currently we have to fix three things:
1. Node Count should be grayed out when Auto Scaling is enabled
2. Disabling Auto Scaling for Node Pools should remove also `maxCount` and `minCount` fields
3. I know that it is unrelated but it is so small that can be fixed quick. We would like to change placeholder for `Cluster Resource Group` from `docker-machine` to `e.g. aks-resource-group`

**Comment by prabalsharma** [comment]
looks like PR is still in review https://github.com/rancher/aks-operator/pull/27

**Comment by prabalsharma** [comment]
tested on 7a2fc74, aks-operator:v1.0.1-rc11

**Comment by mjura** [comment]
Bug is fixed in v1.0.1-rc14
https://github.com/rancher/rancher/pull/33413

**Comment by prabalsharma** [comment]
> Bug is fixed in v1.0.1-rc14
> #33413

@mjura what is this fix for? This issue is already closed. Do you want me to re-open an re-test this issue or you commented on it by mistake?
